data-prepper-plugins/grok-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepper.java

```diff
@@ -56,6 +57,7 @@
 import java.util.stream.Collectors;
 
 
+@SingleThread
```

Comment:
So looks like the ProcessWorkers was not creating as many threads as number of workers defined in the pipeline configuration.

https://github.com/opensearch-project/data-prepper/blob/6d597af802bf358a690fc9a653f0a97d5f90e5bc/data-prepper-core/src/main/java/com/amazon/dataprepper/parser/PipelineParser.java#L133-L140

Reply:
@dinujoh , By default, one instance of each `Prepper` is created and shared across the worker threads. I think that this is probably good for many `Prepper`s. However, each Grok instance creates a single-threaded `ExecutorService`.

Because the single instance is shared between worker threads, there is only one `ExecutorService` for all the threads. This is a single-thread executor service. Thus, each thread will be blocked by that single thread.

Note that this `ExecutorService` is used only when the `time_out` is non-zero.